spec/lib/solidus_graphql_api/queries/products_query_spec.rb

```diff
@@ -3,9 +3,54 @@
 require 'spec_helper'
 
 RSpec.describe SolidusGraphqlApi::Queries::ProductsQuery do
-  subject { described_class.new.call }
+  subject do
+    described_class.new(user: user, pricing_options: pricing_options).call(query: query)
+  end
 
-  let!(:products) { create_list(:product, 2) }
+  let(:user) { create :user }
+  let(:pricing_options) { Spree::Config.pricing_options_class.new }
+  let(:query) { {} }
 
-  it { is_expected.to match_array(products) }
+  context 'when products do not exist' do
```

Comment:
We test the real behavior of this method into the integration spec, and I think we can simplify these specs by mocking the request to the Solidus search class.